### ISSUE TITLE
Potential bug in WorldModel.possible_successors method

### ISSUE THREAD

**Issue by Jay5th** [pik-gane/satisfia#12]
Jobst and I discovered this potential bug during a meeting.

Consider the following method from the  WorldModel class:

![WorldModel_bug](https://github.com/user-attachments/assets/2be23e39-246f-4a3e-9928-a96f4d9b56a6)

This is the current implementation of the possible_successors method. I may be confused, but it looks like the recursive call could lead to incorrect behavior. For example, say we are at a terminal state, and I don't provide an action. In fact, since the state is terminal, there are no possible actions and no possible successors. In this case, the method returns the current answer, which is an empty set. However, if instead we start at a state immediately prior to a terminal state without providing an action, the method will pull actions out of the space of possible actions and again call possible_successors. Note that this recursive call does not modify the state or n_samples. Then, within this recursive call, the method takes the action that it was given and again calls possible_successors. This time, not even the action has been modified, so the method calls itself using the exact same inputs. 

Again, I may be confused, so please check my logic.

**Comment by mensch72** [comment]
yes, i think the method needs to be basically reverted to its original version which uses transition_distribution().keys()

Am 16. Juli 2024 17:04:53 MESZ schrieb Jay5th ***@***.***>:
